posydon/visualization/plot_pop.py

```diff
@@ -0,0 +1,254 @@
+                         alpha=alpha, range=range, bins=bins)
+    if df_observable is not None:
+        # normalise weight, for GRB this will conserve GRB1/GRB2 ratio
+        df_observable['weight'] /= sum(df_observable['weight'])
+        if channel is not None:
+            sel = (df_observable['weight'] > 0) & (df_observable['channel'] == channel)
+            if channel not in title:
+                title += f'\n{channel}'
+        else:
+            sel = df_observable['weight'] > 0
+        if isinstance(x, str):
+            if pop == 'GRB':
+                sel_tmp = sel & ~df_observable[x].isna()
+            else:
+                sel_tmp = sel
+            values, weights = df_observable.loc[sel_tmp,[x,'weight']].values.T
+            if xlog:
+                mask = values > 0
+                values = np.log10(values[mask])
+                weights = weights[mask]
+            plt.hist(values, weights=weights,
+                     color=COLORS[0],
+                     histtype=u'step', linestyle='--',
+                     alpha=alpha, range=range, bins=bins)
+        elif isinstance(x, list):
+            for i, x_i in enumerate(x):
+                if "S1" in x_i:
+                    label = 'S1'
+                    s = 1
+                elif "S2" in x_i:
+                    label = 'S2'
+                    s = 2
+                else:
+                    label = x_i
+                if pop == 'GRB':
+                    sel_tmp = sel & ~df_observable[x_i].isna() & df_observable[f'GRB{s}']
+                else:
+                    sel_tmp = sel
+                values, weights = df_observable.loc[sel_tmp,[x_i,'weight']].values.T
+                if xlog:
+                    mask = values > 0
+                    values = np.log10(values[mask])
+                    weights = weights[mask]
+                plt.hist(values, weights=weights,
+                         color=COLORS[i], label=label,
+                         histtype=u'step', linestyle='--',
+                         alpha=alpha, range=range, bins=bins)
+    plt.title(title)
+    if ylog:
+        plt.yscale('log')
+    plt.ylabel(r'PDF')
+    try:
+        if isinstance(x, str):
+            if not xlog:
+                plt.xlabel(DEFAULT_LABELS[x][0])
+            else:
+                plt.xlabel(DEFAULT_LABELS[x][1])
+        else:
+            if not xlog:
+                plt.xlabel(DEFAULT_LABELS[x[0]][0])
+            else:
+                plt.xlabel(DEFAULT_LABELS[x[0]][1])
+    except:
+        if isinstance(x, str):
+            if not xlog:
+                plt.xlabel(x)
+            else:
+                plt.xlabel('log10_'+x)
+        else:
+            if not xlog:
+                plt.xlabel(x[0])
+            else:
+                plt.xlabel('log10_'+x[0])
+    if isinstance(x, list):
+        plt.legend(loc=1)
+    if path:
+        plt.savefig(path)
+    if show:
+        plt.show()
```

Comment:
Sometimes I like to return the ax object (when there is one) so that I can make edits or overplot something. Something to consider eventually maybe

Reply:
Makes sense, for the other plotting libraries I implemented it when it was needed. I suggest that the first time someone needs it should implement it.